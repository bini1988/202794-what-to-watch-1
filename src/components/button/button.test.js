import React from "react";
import renderer from "react-test-renderer";
import Button from "./button";

const Nop = () => {};

it(`Button correctly renders default markup`, () => {
  const tree = renderer.create(
      <Button label="btn label" onClick={Nop}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Button correctly renders play mode markup`, () => {
  const tree = renderer.create(
      <Button label="btn label" mode="play" onClick={Nop}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Button correctly renders list mode markup`, () => {
  const tree = renderer.create(
      <Button label="btn label" mode="list" onClick={Nop}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Button correctly renders inlist mode markup`, () => {
  const tree = renderer.create(
      <Button label="btn label" mode="inlist" onClick={Nop}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
