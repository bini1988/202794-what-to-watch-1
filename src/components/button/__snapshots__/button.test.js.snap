// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button correctly renders default markup 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  <span>
    btn label
  </span>
</button>
`;

exports[`Button correctly renders inlist mode markup 1`] = `
<button
  className="btn btn--inlist"
  onClick={[Function]}
  type="button"
>
  <svg
    height={14}
    viewBox="0 0 18 14"
    width={18}
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    btn label
  </span>
</button>
`;

exports[`Button correctly renders list mode markup 1`] = `
<button
  className="btn btn--list"
  onClick={[Function]}
  type="button"
>
  <svg
    height={20}
    viewBox="0 0 19 20"
    width={19}
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    btn label
  </span>
</button>
`;

exports[`Button correctly renders play mode markup 1`] = `
<button
  className="btn btn--play"
  onClick={[Function]}
  type="button"
>
  <svg
    height={19}
    viewBox="0 0 19 19"
    width={19}
  >
    <use
      xlinkHref="#play-s"
    />
  </svg>
  <span>
    btn label
  </span>
</button>
`;
